src/nnet3/nnet-utils.cc

```diff
@@ -40,6 +40,106 @@ int32 NumInputNodes(const Nnet &nnet) {
   return ans;
 }
 
+int32 GetTModulusForIvector(const Nnet &nnet) {
+  std::vector<std::string> name;
+  name.push_back("ivector");
+  std::vector<int32> value;
+  value.resize(name.size(), 0);
+  for (int32 i = 0; i < nnet.NumNodes(); i++) {
+    const NetworkNode &node = nnet.GetNode(i);
+    // skip output node to circumvent the problem when a recurrent output 
+    // is "label delayed" and thus also contain keyword "Offset", which is not
+    // what we are trying to look at for our "offset"  
+    if (node.node_type == kDescriptor) {
+      for (int32 p = 0; p < node.descriptor.NumParts(); p++) {
+        const SumDescriptor &this_part = node.descriptor.Part(p);
+        IntoSumDescriptor(nnet, this_part, name, &value);
+      }
+    }
+  }
+  return value[0]; 
+}
+
+void IntoSumDescriptor(const Nnet &nnet,
+                       const SumDescriptor &this_descriptor,
+                       const std::vector<std::string> &node_names,
+                       std::vector<int32> *values) {
+  const OptionalSumDescriptor *ptr_optional =
+      dynamic_cast<const OptionalSumDescriptor*>(&this_descriptor);
+  if (ptr_optional != NULL) {
+    IntoSumDescriptor(nnet, *(ptr_optional->src_), node_names, values);
+    return;
+  }
+  const SimpleSumDescriptor *ptr_simple =
+      dynamic_cast<const SimpleSumDescriptor*>(&this_descriptor);
+  if (ptr_simple != NULL) {
+    IntoForwardingDescriptor(nnet, *(ptr_simple->src_), node_names, values);
+    return;
+  }
+  const BinarySumDescriptor *ptr_binary =
+      dynamic_cast<const BinarySumDescriptor*>(&this_descriptor);
+  if (ptr_binary != NULL) {
+    IntoSumDescriptor(nnet, *(ptr_binary->src1_), node_names, values);
+    IntoSumDescriptor(nnet, *(ptr_binary->src2_), node_names, values);
+    return;
+  }
+  KALDI_ERR << "Unidentified SumDescriptor.";
```

Comment:
the function needs documentation- it's not clear to me what it's supposed to do and the name provides no clue.